src/main/java/org/folio/service/financedata/FinanceDataService.java

```diff
@@ -0,0 +1,103 @@
+package org.folio.service.financedata;
+
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+import java.util.List;
+
+import io.vertx.core.Future;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.Fund;
+import org.folio.rest.jaxrs.model.FyFinanceData;
+import org.folio.rest.jaxrs.model.FyFinanceDataCollection;
+import org.folio.rest.jaxrs.model.Tags;
+import org.folio.rest.persist.DBConn;
+import org.folio.service.budget.BudgetService;
+import org.folio.service.fund.FundService;
+
+public class FinanceDataService {
+  private static final Logger logger = LogManager.getLogger();
+
+  private final FundService fundService;
+  private final BudgetService budgetService;
+
+  public FinanceDataService(FundService fundService, BudgetService budgetService) {
+    this.fundService = fundService;
+    this.budgetService = budgetService;
+  }
+
+  public Future<Void> update(FyFinanceDataCollection entity, RequestContext requestContext) {
+    if (CollectionUtils.isEmpty(entity.getFyFinanceData())) {
+      return Future.succeededFuture();
+    }
+    var dbClient = requestContext.toDBClient();
+    return dbClient
+      .withTrans(conn -> {
+        var updateFundFuture = processFundUpdate(entity, conn);
+        var updateBudgetFuture = processBudgetUpdate(entity, conn);
```

Comment:
some more update is needed here. During updating budgest we are creating allocation transactions and need to set description and transaction tag for each transaction.
Transaction description, transaction tags are described in related UI story https://folio-org.atlassian.net/browse/UIF-527
and this miro diagram https://miro.com/app/board/uXjVKJnUzfQ=/

Reply:
or we will set this in mod-finance PR because allocation transactions are created here

Reply:
resolving comment, will be handled in mod-finance module